app/Http/Controllers/ProjectController.php

```diff
@@ -3,20 +3,25 @@
 namespace App\Http\Controllers;
 
 use App\Project;
+use App\ProjectComment;
+use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
-use Illuminate\Support\Facades\Request;
+use Session;
+//use Illuminate\Support\Facades\Request;
 
 class ProjectController extends Controller
 {
     /**
      * Renders the Project View for the given id
      *
-     * @param $id project id for lookup
+     * @param $project project id cast to Project
      * @return The project page view
      */
-    public function view($id)
+    public function view(Project $project)
```

Comment:
Oh, so this does an implicit conversion to a project model? That's super neat. Good to know :+1: